x/mongo/driver/compression_test.go

```diff
@@ -40,3 +42,120 @@ func TestCompression(t *testing.T) {
 		})
 	}
 }
+
+func TestZstdWindowSize(t *testing.T) {
+	tests := []struct {
+		inputSize  int
+		level      zstd.EncoderLevel
+		windowSize int
+	}{
+		{
+			inputSize:  0,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 1024,
+		},
+		{
+			inputSize:  512,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 1024,
+		},
+		{
+			inputSize:  512000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 524288,
+		},
+		{
+			inputSize:  16000000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 16777216,
+		},
+		{
+			inputSize:  32000000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 16777216,
+		},
+		{
+			inputSize:  4000000,
+			level:      zstd.SpeedFastest,
+			windowSize: 4194304,
+		},
+		{
+			inputSize:  8000000,
+			level:      zstd.SpeedFastest,
+			windowSize: 4194304,
+		},
+		{
+			inputSize:  8000000,
+			level:      zstd.SpeedDefault,
+			windowSize: 8388608,
+		},
+		{
+			inputSize:  16000000,
+			level:      zstd.SpeedDefault,
+			windowSize: 8388608,
+		},
+		{
+			inputSize:  16000000,
+			level:      zstd.SpeedBetterCompression,
+			windowSize: 16777216,
+		},
+		{
+			inputSize:  32000000,
+			level:      zstd.SpeedBetterCompression,
+			windowSize: 16777216,
+		},
+		{
+			inputSize:  32000000,
+			level:      zstd.SpeedBestCompression,
+			windowSize: 33554432,
+		},
+		{
+			inputSize:  64000000,
+			level:      zstd.SpeedBestCompression,
+			windowSize: 33554432,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("size %d level %v", test.inputSize, test.level), func(t *testing.T) {
+			windowSize := calcZstdWindowSize(test.inputSize, test.level)
+			assert.Equal(t, test.windowSize, windowSize)
+		})
+	}
+}
+
+func BenchmarkCompression(b *testing.B) {
```

Comment:
Optional: Consider renaming this to `BenchmarkCompressPayload` because `CompressPayload` is the specific function that's being benchmarked.